#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

			     BASH PATCH REPORT
			     =================

Bash-Release: 3.2
Patch-ID: bash32-040

Bug-Reported-by:	John McCabe-Dansted
Bug-Reference-ID:
Bug-Reference-URL:	https://bugs.launchpad.net/ubuntu/+source/bash/+bug/202885

Bug-Description:

When using the `set' builtin to list all shell variables, the shell uses
the wrong variable when computing the length of a variable's value.

Patch:

*** ../bash-3.2-patched/array.c	2007-03-24 14:51:03.000000000 -0400
--- array.c	2008-08-17 13:07:04.000000000 -0400
***************
*** 684,688 ****
  		valstr = element_value (ae) ? sh_double_quote (element_value(ae))
  					    : (char *)NULL;
! 		elen = STRLEN (indstr) + 8 + STRLEN (valstr);
  		RESIZE_MALLOCED_BUFFER (result, rlen, (elen + 1), rsize, rsize);
  
--- 809,813 ----
  		valstr = element_value (ae) ? sh_double_quote (element_value(ae))
  					    : (char *)NULL;
! 		elen = STRLEN (is) + 8 + STRLEN (valstr);
  		RESIZE_MALLOCED_BUFFER (result, rlen, (elen + 1), rsize, rsize);
  
*** ../bash-3.2/patchlevel.h	Thu Apr 13 08:31:04 2006
--- patchlevel.h	Mon Oct 16 14:22:54 2006
***************
*** 26,30 ****
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 39
  
  #endif /* _PATCHLEVEL_H_ */
--- 26,30 ----
     looks for to find the patch level (for the sccs version string). */
  
! #define PATCHLEVEL 40
  
  #endif /* _PATCHLEVEL_H_ */
